src/dialects/oracle/connection-manager.js

```diff
@@ -58,32 +58,19 @@ export class OracleConnectionManager extends AbstractConnectionManager {
    * @returns {Promise<Connection>}
    * @private
    */
-  checkConfigObject(config) {
-    // A connectString should be defined
-    if (config.database.length === 0) {
-      let errorToThrow =
-        'The database cannot be blank, you must specify the database name (which correspond to the service name';
-      errorToThrow += '\n from tnsnames.ora : (HOST = mymachine.example.com)(PORT = 1521)(SERVICE_NAME = orcl)';
-      throw new Error(errorToThrow);
+  buildConnectString(config) {
+    if (!config.host || config.host.length === 0)
+      return config.database;
+    let connectString = config.host;
+    if (config.port && config.port > 0) {
+      connectString += `:${config.port}`;
+    } else {
+      connectString += ':1521';
     }
-
-    if (!config.host || config.host.length === 0) {
-      throw new Error('You have to specify the host');
-    }
-
-    // The connectString has a special format, we check it
-    // ConnectString format is : host:[port]/service_name
-    if (config.database.indexOf('/') === -1) {
-      let connectString = config.host;
-
-      if (config.port && config.port !== 0) {
-        connectString += `:${config.port}`;
-      } else {
-        connectString += ':1521'; //Default port number
-      }
+    if (config.database && config.database.length > 0) {
       connectString += `/${config.database}`;
```

Comment:
no need to urlEncode config.database?

Reply:
Its not needed. config.database is servicename and can be handled by node-oracledb driver.

Reply:
just to add, the connectString will have the uri format built from the user provided database and options. It will be like
`'adb.oracle.com:1521/XEPDB1'`

Reply:
Is it possible for `database` to be named something like `test/test`? I only had a few minutes to try this morning before starting work but I set the `ORACLE_DATABASE` env variable in `gvenzl/oracle-xe` to `test/test` and it seemed to work, but sequelize wouldn't connect to it because `adb.oracle.com:1522/test/test` is not valid

Anyway it's not caused by this PR so I won't block because of it. Feel free to resolve this conversation once you've seen it :)